lib/test-ast.js

```diff
@@ -0,0 +1,54 @@
+'use strict';
+
+const fs = require('fs');
+const AST = require('recast');
+
+module.exports = ({startLineNumber, title}, filePath) => {
+	const fileAst = AST.parse(tryReadFile(filePath));
```

Comment:
Turns out this code runs for every test declaration… so we'll definitely have to cache this. Maybe using `mem`? Or `runner` could parse in advance? In fact we'd probably want to cache the tree traversal.

Reply:
Yeah, that’s what I tried to tell you before.

Sounds good!

Reply:
Yea… sorry about that.

Reply:
Just to make sure: When you say _”cache the tree traversal”_, you mean the parsed test AST, right?

Also, have you decided on whether to have this done up front by `runner.js` or on-the-fly?

Reply:
If we know, in the runner, that we're selecting by lines *in this file*, then we can analyze up front and build a map of declaration start and end lines, indexed by the start line.

Reply:
Aye, I'll give it a try.

Reply:
I'm skipping this one until it's been determined start line number is unique enough.

Reply:
I went ahead and did this as well, according to the new approach